ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -59,12 +59,19 @@ export class JobResourcesComponent implements OnInit {
       }
 
       if (this.job.extensions.logs) {
-        for (let file of Object.keys(this.job.extensions.logs)) {
-          this.readResourceFile(file);
-          let tabId = 'log-' + file;
-          this.tabIds.push(tabId);
-          this.tabTitles.set(tabId, file);
-        }
+        const files = Object.keys(this.job.extensions.logs);
+        Promise.all(files.map(file => this.readResourceFile(file)))
+          .then(entries => {
+            // Sort log files by the file name (tuples will be converted to
+            // string and compared).
+            for (let [file, data] of entries.filter(l => !!l).sort()) {
```

Comment:
nit: not a fan of `l` as a var name, would prefer `e` here, or anything else.

Reply:
Right good call, fixed.